crypto/crypto_aesctr.c

```diff
@@ -93,14 +99,19 @@ crypto_aesctr_init(const struct crypto_aes_key * key, uint64_t nonce)
 static inline void
 crypto_aesctr_stream_cipherblock_generate(struct crypto_aesctr * stream)
 {
-	uint8_t pblk[16];
 
-	/* Prepare nonce and counter. */
-	be64enc(pblk, stream->nonce);
-	be64enc(pblk + 8, stream->bytectr / 16);
+	/* Sanity check. */
+	assert (stream->bytectr % 16 == 0);
+
+	/* Prepare counter. */
+	stream->pblk[15]++;
+	if (stream->pblk[15] == 0) {
+		/* If there was an overflow, write the full value. */
```

Comment:
"If incrementing the least significant byte results in it wrapping, re-encode the complete 64-bit value."